src/Core/Services/StateService.cs

```diff
@@ -67,6 +71,28 @@ public async Task SetActiveUserAsync(string userId)
             await SetPreAuthEnvironmentUrlsAsync(await GetEnvironmentUrlsAsync());
         }
 
+        public async Task CheckExtensionActiveUserAndSwitchIfNeededAsync()
+        {
+            var extensionUserId = await GetExtensionActiveUserIdFromStorageAsync();
+            if (string.IsNullOrEmpty(extensionUserId))
+            {
+                return;
+            }
+
+            if (_state?.ActiveUserId == extensionUserId)
```

Comment:
Any reason not to use `GetActiveUserIdAsync()` here instead of `_state?.ActiveUserId`?

Reply:
I don't want to get the user id from storage. If the extension changed the user then it also changed the active user id on storage thus the validation wouldn't make any sense because the storage and the extension user id would be the same.
What I want is to check whether the in memory user id (that is the one in place when sending the app to background to open another app that opens the extension) is different from the one that the extension changed.